currency/manager.go

```diff
@@ -188,6 +188,38 @@ func (p *PairsManager) StorePairs(a asset.Item, pairs Pairs, enabled bool) error
 	return nil
 }
 
+// EnsureOnePairEnabled not all assets have pairs, eg options
+// search for an asset that does and enable one if none are enabled
+// error if no currency pairs found for an entire exchange
+func (p *PairsManager) EnsureOnePairEnabled() error {
+	p.m.Lock()
+	defer p.m.Unlock()
+	for i := range p.Pairs {
+		if p.Pairs[i].AssetEnabled == nil ||
+			!*p.Pairs[i].AssetEnabled ||
+			len(p.Pairs[i].Available) == 0 {
+			continue
+		}
+		if len(p.Pairs[i].Enabled) > 0 {
+			return nil
+		}
+	}
+	for i := range p.Pairs {
+		if p.Pairs[i].AssetEnabled == nil ||
+			!*p.Pairs[i].AssetEnabled ||
+			len(p.Pairs[i].Available) == 0 {
+			continue
+		}
+		rp, err := p.Pairs[i].Available.GetRandomPair()
+		if err != nil {
+			return err
+		}
+		p.Pairs[i].Enabled = p.Pairs[i].Enabled.Add(rp)
+		return nil
+	}
```

Comment:
That doesn't achieve the same result. 
Let's take the following scenario: 
- I have 2 assets: futures and spot.
- I have futures with zero enabled and spot with one
- Based on my code, it will not action anything because there is a pair enabled in spot
- Based on your suggested loop, this will enable a random pair in futures

This function is only wanting to enable one asset pair out of everything available for an exchange so it needs to finish the for loop before deciding to enable something

Reply:
That makes sense, sorry to be a pain but can you add a test for that case? My changes above passed the test. And I am sure I would forget this conversation ever happening and change it later down the track.

Reply:
Commit [a74dfc0](https://github.com/thrasher-corp/gocryptotrader/pull/1159/commits/a74dfc01ba4f1e8d73b6d53ead1cd6ac641230cb) would have demonstrated why your loop simplification wouldn't have worked if you're interested as you didn't provide what test you ran passed. It was a very rare scenario, but I don't want to focus on this to block the rest of the PR and I've removed it all

Reply:
I just ran the test that was on the branch provided for `EnsureOnePairEnabled` and it passed. 🤷 Maybe I was wrong.